subsys/dfu/boot/mcuboot.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2016 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCUBOOT_H__
+#define __MCUBOOT_H__
+
+#include "boot/mcuboot_constraints.h"
+
+/**
+ * @brief Marks the image in slot 0 as confirmed. The system will continue booting
+ * into the image in slot 0 until told to boot from a different slot.
```

Comment:
Not quite accurate. There are two ways of doing this:
1. One-shot upgrade. Write the new image in slot 1, add image magic, and set image_ok. The bootloader will move to slot 0, and boot it from then on.
2. Trial upgrade. Write the image in slot 1, and add image magic. Bootloader will swap images, and boot slot 0. If image works, it should set image_ok in slot 0.

However, it needs to be clear that this function really isn't part of the DFU or upgrade, but needs to be done on every boot.